source/src/protocols/cyclic_peptide_predict/SimpleCycpepPredictApplication.cc

```diff
@@ -1761,6 +1791,18 @@ SimpleCycpepPredictApplication::run() const {
 			set_up_terminal_disulfide_variants( pose ); //Add disulfide variants, if we're doing disulfide cyclization.
 		} else if ( cyclization_type() == SCPA_thioether_lariat ) {
 			thioether_sidechain_index = set_up_terminal_thioether_lariat_variants( pose ); //Add thioether lariat variant to the N-terminus, if we're doing that type of cyclization.
+		} else if ( cyclization_type() == SCPA_lanthipeptide ) {
+			//need to call for all lan and melan resi pairs
+			if ( lanthionine_positions_.size() > 0 ) {
+				for ( core::Size i=1, imax=lanthionine_positions_.size(); i<=imax; ++i ) { //Loop through all sets of doubles of residues.
+					debug_assert(lanthionine_positions_[i].size() == 2); //Should always be true.
+					protocols::cyclic_peptide::crosslinker::set_up_lanthionine_variants(*pose, lanthionine_positions_[i][1], lanthionine_positions_[i][2]);
+				}
+				core::pose::add_lower_terminus_type_to_pose_residue( *pose, 1 );
+				core::pose::add_upper_terminus_type_to_pose_residue( *pose, pose->size() );
+			} else {
+				runtime_assert_string_msg( lanthionine_positions_.size() > 0 , "Error in protocols::cyclic_peptide_predict::SimpleCycpepPredictApplication lanthionine positions are empty");
```

Comment:
I'll mention that you probably don't need to have an explicit `if()` test and then runtime_assert with the same condition. If the assumption is that you'll only reach this point if you have a valid  lanthionine_positions_ setting, you can just do the runtime_assert, and continue with the "passed" code at the same level.